### PR TITLE
[IMP] web: add cumulative line chart

### PR DESCRIPTION
**Pull request by fdardenne**
The cumulative option on line chart is useful for displaying
cumulative information graphically.

task-id: 2146487

### CODE REVIEW

File: addons/web/static/src/views/graph/graph_controller.xml [hunk 1]
```diff
@@ -38,6 +38,10 @@
                 t-on-click="toggleStacked"
                 t-att-class="{ active: model.metaData.stacked }"
             />
+            <button class="btn btn-secondary fa fa-signal o_graph_button" data-tooltip="Cumulative" aria-label="Cumulative"
+                t-on-click="toggleCumulated"
+                t-att-class="{ active: model.metaData.cumulated }"
```
Comment (FrancoisGe) [inline addons/web/static/src/views/graph/graph_controller.xml:43]: why display the button if you can't click it? t-if = model.metaData.cumulated ?

File: addons/web/static/src/views/graph/graph_view.js [hunk 2]
```diff
@@ -43,6 +43,7 @@ export const graphView = {
                 order: archInfo.order || null,
                 resModel: resModel,
                 stacked: "stacked" in archInfo ? archInfo.stacked : true,
+                cumulated: "cumulated" in archInfo ? archInfo.cumulated : false,
```
Comment (FrancoisGe) [inline addons/web/static/src/views/graph/graph_view.js:46]: Two shorter way to do the same thing
!!archInfo.cumulated
or
Boolean(archInfo.cumulated)

File: addons/web/static/src/views/graph/graph_arch_parser.js [hunk 3]
```diff
@@ -22,6 +22,9 @@ export class GraphArchParser extends XMLParser {
                     if (node.hasAttribute("stacked")) {
                         archInfo.stacked = archParseBoolean(node.getAttribute("stacked"));
                     }
+                    if (node.hasAttribute("cumulated")) {
+                        archInfo.cumulated = archParseBoolean(node.getAttribute("cumulated"));
+                    }
```
Comment (FrancoisGe) [inline addons/web/static/src/views/graph/graph_arch_parser.js:27]: You could replace cumulated: "cumulated" in archInfo ? archInfo.cumulated : false,
By archInfo.cumulated = node.getAttribute("cumulated") ? archParseBoolean(node.getAttribute("cumulated")) : false

File: addons/web/static/src/views/graph/graph_renderer.js [hunk 4]
```diff
@@ -348,6 +348,13 @@ export class GraphRenderer extends Component {
             if (stacked) {
                 dataset.backgroundColor = hexToRGBA(dataset.borderColor, 0.4);
             }
+            if (cumulated) {
+                let accumulator = 0;
+                dataset.data = dataset.data.map((value) => {
+                    accumulator += value;
+                    return accumulator;
```
Comment (FrancoisGe) [inline addons/web/static/src/views/graph/graph_renderer.js:355]: You should use .reduce :)

Comment (SimonGenin) [reply addons/web/static/src/views/graph/graph_renderer.js:355]: I though so but no, it's a map and reduce

File: addons/web/static/src/views/graph/graph_controller.xml (hunk 1 above)
Comment (SimonGenin) [reply addons/web/static/src/views/graph/graph_controller.xml:43]: active is not the same as clickable @FrancoisGe ? It just change its style to underline it's active

**Comment by FrancoisGe** [comment]
@robodoo r+

**Comment by FrancoisGe** [comment]
@robodoo rebase-ff

**Pull request closed without merging**